packages/docregistry/src/registry.ts

```diff
@@ -1182,7 +1182,11 @@ export namespace DocumentRegistry {
      *
      * @returns A new document model.
      */
-    createNew(languagePreference?: string, collaborationEnabled?: boolean): T;
+    createNew(
+      languagePreference?: string,
+      sharedModel?: ISharedDocument,
+      collaborationEnabled?: boolean
+    ): T;
```

Comment:
Could you please change the signature to use a object rather than a list of parameters? You may call it `DocumentRegistry.IModelOptions` for example.

It will ease future evolution if needed.

```suggestion
    createNew(options?: {
      languagePreference?: string,
      sharedModel?: ISharedDocument,
      collaborationEnabled?: boolean
    }): T;
```